src/main/java/com/faforever/client/map/generator/MapGeneratorService.java

```diff
@@ -124,26 +152,62 @@ protected String queryNewestVersion() {
     JsonElement jsonElement = new JsonParser().parse(response.getBody());
     JsonObject mainObject = jsonElement.getAsJsonObject();
 
-    return mainObject.get("tag_name").getAsString();
+    return new ComparableVersion(mainObject.get("tag_name").getAsString());
+  }
+
+  @VisibleForTesting
+  @Cacheable(CacheNames.MAP_GENERATOR)
+  public ComparableVersion queryMaxSupportedVersion() {
+    ComparableVersion version = new ComparableVersion("");
+    ComparableVersion minVersion = new ComparableVersion(String.valueOf(clientProperties.getMapGenerator().getMinSupportedMajorVersion()));
+    ComparableVersion maxVersion = new ComparableVersion(String.valueOf(clientProperties.getMapGenerator().getMaxSupportedMajorVersion() + 1));
+
+    RestTemplate restTemplate = new RestTemplate();
+
+    LinkedMultiValueMap<String, String> headers = new LinkedMultiValueMap<>();
+    headers.add("Accept", "application/vnd.github.v3+json");
+    HttpEntity<String> entity = new HttpEntity<>(null, headers);
+
+    ResponseEntity<String> response = restTemplate.exchange(clientProperties.getMapGenerator().getQueryVersionsUrl(), HttpMethod.GET, entity, String.class);
+    JsonElement jsonElement = new JsonParser().parse(response.getBody());
+    for (JsonElement element : jsonElement.getAsJsonArray()) {
+      JsonObject mainObject = element.getAsJsonObject();
+      version.parseVersion(mainObject.get("tag_name").getAsString());
+      if (version.compareTo(maxVersion) < 0 && minVersion.compareTo(version) < 0) {
+        return version;
+      }
+    }
```

Comment:
Is there a reason not to use jackson? You can configure a pojo to ignore unknown fields.

Reply:
Yeah I am going to say the reason is that I have no idea what a jackson is.

Reply:
Oh. Jackson is the **json** de/serializer in Spring (https://github.com/FasterXML/jackson). Spring uses it by default e.g. if you call a RestTemplate, you tell Spring it needs to return a type T and Jackson will convert it to type T.
(The client also uses Gson which is another json de/serializer by Google, but in the long run, everything should use Jackson).

You can get it via Dependency Injection. The class you need to ask for is an `ObjectMapper`.

Reply:
Alternatively you can ask the RestTemplate directly to return you the pojo you need.

Reply:
Have it return from RestTemplate directly